File: desktop/src/api/modules/params.ts
import { apiGet, apiPost, apiPut } from "@/api/client";
import type {
  AdvancedModelParams,
  DynamicModelParams,
  FraudModelParams,
  ParamsOverviewResponse,
  SaveMessageResponse,
  UpdateCheckResponse,
} from "@/types/params";

export function fetchParamsOverview() {
  return apiGet<ParamsOverviewResponse>("/api/params/overview");
}

export function saveFraudModelParams(payload: FraudModelParams) {
  return apiPut<SaveMessageResponse>("/api/params/fraud-model", payload);
}

export function saveAdvancedModelParams(payload: AdvancedModelParams) {
  return apiPut<SaveMessageResponse>("/api/params/advanced-model", payload);
}

export function saveDynamicModelParams(payload: DynamicModelParams) {
  return apiPut<SaveMessageResponse>("/api/params/dynamic-model", payload);
}

export function triggerSoftwareUpdate() {
  return apiPost<UpdateCheckResponse>("/api/params/actions/software-update", {});
}

export function triggerModelUpdate() {
  return apiPost<UpdateCheckResponse>("/api/params/actions/model-update", {});
}

export function triggerParameterUpdate() {
  return apiPost<UpdateCheckResponse>("/api/params/actions/parameter-update", {});
}

File: server/app/api/routes/params.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from app.api.routes.auth import get_current_user
from app.features.params.schemas import (
    AdvancedModelParamsUpdateRequest,
    DynamicModelParamsUpdateRequest,
    FraudModelParamsUpdateRequest,
    ParamsOverviewResponse,
    SaveMessageResponse,
    UpdateCheckResponse,
)
from app.features.params.service import service

router = APIRouter()


@router.get("/overview", response_model=ParamsOverviewResponse)
async def overview(current_user=Depends(get_current_user)):
    return service.overview()


@router.put("/fraud-model", response_model=SaveMessageResponse)
async def save_fraud_model(payload: FraudModelParamsUpdateRequest, current_user=Depends(get_current_user)):
    return service.save_fraud_model(payload)


@router.put("/advanced-model", response_model=SaveMessageResponse)
async def save_advanced_model(payload: AdvancedModelParamsUpdateRequest, current_user=Depends(get_current_user)):
    return service.save_advanced_model(payload)


@router.put("/dynamic-model", response_model=SaveMessageResponse)
async def save_dynamic_model(payload: DynamicModelParamsUpdateRequest, current_user=Depends(get_current_user)):
    return service.save_dynamic_model(payload)


@router.post("/actions/software-update", response_model=UpdateCheckResponse)
async def update_software(current_user=Depends(get_current_user)):
    return service.update_software_version()


@router.post("/actions/model-update", response_model=UpdateCheckResponse)
async def update_model(current_user=Depends(get_current_user)):
    return service.update_model_version()


@router.post("/actions/parameter-update", response_model=UpdateCheckResponse)
async def update_parameter(current_user=Depends(get_current_user)):
    return service.update_parameter_version()
